contracts/erc3643/src/hooks/call_hook.rs

```diff
@@ -0,0 +1,70 @@
+use multiversx_sc_modules::transfer_role_proxy::PaymentsVec;
+
+use super::hook_type::{Hook, ErcHookType};
+
+multiversx_sc::imports!();
+
+#[multiversx_sc::module]
+pub trait CallHookModule {
+    fn call_hook(
```

Comment:
Nope, can't be done with what we have now. The function itself would work fine, but the storage mapper cannot contain interface types. It's why I copy-paste this everywhere instead of simply importing.